src/lib/core/Optional.h

```diff
@@ -157,6 +157,12 @@ class Optional
     }
 
     /** Gets the current value of the optional. Valid IFF `HasValue`. */
+    T & Value()
```

Comment:
comment was moved by adding code. Should the other method also have the same comment explaining its use?

Why was the const version not sufficient?

Reply:
Commend added for both `Value` function.

The non-const variant allow one edit the stored value in-place. This is also provided by `std::optional`